pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1410,7 +1414,14 @@ func getCPUTopology(vmi *v1.VirtualMachineInstance) *CPUTopology {
 	if vmiCPU == nil || (vmiCPU.Cores == 0 && vmiCPU.Sockets == 0 && vmiCPU.Threads == 0) {
 		//if cores, sockets, threads are not set, take value from domain resources request or limits and
 		//set value into sockets, which have best performance (https://bugzilla.redhat.com/show_bug.cgi?id=1653453)
-		resources := vmi.Spec.Domain.Resources
+
+		// libvirt sets the `multiqueue` flag based on the number of vCPUs
+		// requested. As such, when we request multi-queue, we must help
+		// libvirt to honor our previous request.
+		if IsVMISpecRequestingNetworkMultiqueue(vmiSpec) {
+			sockets = uint32(2)
+		}
+		resources := vmiSpec.Domain.Resources
```

Comment:
I think this is problematic to modify guest topology. if nothing was specified it should remain as 1 vCPU. 
We also calculate the memory overhead based on the number of vCPUs - changing this value here will make it not to be considered.

The 2 lines below are also very problematic. I think we need to remove this. I guess not in this PR.

Reply:
I'm not sure what I proposed is a good idea; but, a user that just sets multi-queue and does not set the number of CPUs will have the VM instantiation fail (well, we'll indicate what is wrong with the request, but still ...).

That is the only thing I was trying to prevent.

I'll update the PR to use a single queue then.

Reply:
I don't see why this PR should change converter.go at all.
The admitter rejecting invalid combos should be enough no?

Reply:
I'm unsure if this was a good idea, and you both think it is not; that settles it :) 

Getting rid of it.

Reply:
Done.

Reply:
> I don't see why this PR should change converter.go at all.

I need to touch just a little bit so I can re-use the same function we're using everywhere to calculate the vCPU number, since in the admitter I do not have access to the vmi object, but to the vmi spec. 

> The admitter rejecting invalid combos should be enough no?